File: ServiceApp/src/main/java/com/example/ServiceApp/dto/InterventionSheetDto.java
package com.example.ServiceApp.dto;

import com.example.ServiceApp.entity.Customer;
import com.example.ServiceApp.entity.Employee;
import com.example.ServiceApp.entity.Equipment;
import com.example.ServiceApp.entity.TypeOfIntervention;
import lombok.*;
import org.springframework.boot.autoconfigure.web.WebProperties;

import java.time.LocalDate;
import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class InterventionSheetDto {
    private Long id;
    private TypeOfIntervention typeOfIntervention;
    private Long equipmentId;
    private String serialNumber;
    private LocalDate dateOfIntervention;
    private LocalDate dateOfExpireWarranty;
    private Integer yearsOfWarranty;
    private Long customerId;
    private Long employeeId;
    private String noticed;
    private String fixed;
    private String engineerNote;
    private String customerName;
    private String equipmentName;
    private String employeeName;
    private String signatureBase64;
}

File: service-interface/src/app/dtos/interventionSheetDto.ts
export class InterventionSheetDto {
  id?: number;
  typeOfIntervention?: string;
  equipmentId?: number;
  serialNumber?: string;
  dateOfIntervention?: string;
  dateOfExpireWarranty?: string
  yearsOfWarranty?: number;
  customerId?: number;
  employeeId?: number;
  noticed?: string;
  fixed?: string;
  engineerNote?: string;
  customerName?: string;
  employeeName?: string;
  equipmentName?: string;
  signatureBase64?: string;


  constructor(id: number, typeOfIntervention: string, equipmentId: number, serialNumber: string, dateOfIntervention: string, customerId: number, employeeId: number, noticed: string, fixed: string, engineerNote: string, customerName: string, employeeName: string, equipmentName: string, signatureBase64: string) {
    this.id = id;
    this.typeOfIntervention = typeOfIntervention;
    this.equipmentId = equipmentId;
    this.serialNumber = serialNumber;
    this.dateOfIntervention = dateOfIntervention;
    this.customerId = customerId;
    this.employeeId = employeeId;
    this.noticed = noticed;
    this.fixed = fixed;
    this.engineerNote = engineerNote;
    this.customerName = customerName;
    this.employeeName = employeeName;
    this.equipmentName = equipmentName;
    this.signatureBase64 = signatureBase64;
  }

}
